File: 717.1-bit-and-2-bit-characters/solution.scala
object Solution {
    def isOneBitCharacter(bits: Array[Int]): Boolean = {
        var i = 0
        var one = true
        while (i < bits.length) {
            one = bits(i) == 0
            if (one) i += 1 else i += 2 
        }
        one
    }
}

File: 717.1-bit-and-2-bit-characters/solution.kt
class Solution {
    fun isOneBitCharacter(bits: IntArray): Boolean {
        var i = 0
        var one = true
        while (i < bits.size) {
            one = (bits[i] == 0)
            if (!one) i++
            i++
        }
        return one
    }
}